app/api/groups/withdraw_invitations.robustness.feature

```diff
@@ -0,0 +1,84 @@
+Feature: Withdraw group invitations - robustness
+  Background:
+    Given the database has the following table 'groups':
+      | id  |
+      | 11  |
+      | 12  |
+      | 13  |
+      | 14  |
+      | 21  |
+      | 31  |
+      | 111 |
+      | 121 |
+      | 122 |
+      | 123 |
+      | 131 |
+      | 141 |
+    And the database has the following table 'users':
+      | login | group_id | first_name  | last_name | grade |
+      | owner | 21       | Jean-Michel | Blanquer  | 3     |
+      | user  | 11       | John        | Doe       | 1     |
+      | jane  | 12       | Jane        | Doe       | 1     |
+    And the database has the following table 'group_managers':
+      | group_id | manager_id | can_manage  |
+      | 13       | 21         | memberships |
+      | 13       | 12         | none        |
+    And the database has the following table 'groups_ancestors':
+      | ancestor_group_id | child_group_id | is_self |
+      | 11                | 11             | 1       |
+      | 13                | 13             | 1       |
+      | 13                | 111            | 0       |
+      | 13                | 121            | 0       |
+      | 13                | 123            | 0       |
+      | 14                | 14             | 1       |
+      | 21                | 21             | 1       |
+      | 31                | 31             | 1       |
+      | 111               | 111            | 1       |
+      | 121               | 121            | 1       |
+      | 122               | 122            | 1       |
+      | 123               | 123            | 1       |
+    And the database has the following table 'groups_groups':
+      | id | parent_group_id | child_group_id |
+      | 9  | 13              | 121            |
+      | 10 | 13              | 111            |
+      | 13 | 13              | 123            |
+
+  Scenario: Fails when the user is not a manager of the parent group
+    Given I am the user with id "11"
+    When I send a POST request to "/groups/13/invitations/withdraw?group_ids=31,141,21,11,13"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Fails when the user is a manager of the parent group, but doesn't have enough rights to manage memberships
+    Given I am the user with id "12"
+    When I send a POST request to "/groups/13/invitations/withdraw?group_ids=31,141,21,11,13"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Fails when the user doesn't exist
```

Comment:
This is duty for the middleware, I don't think we should test that in each service.

Reply:
Should I remove the test?

Reply:
No, that's ok.